parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,326 @@
+        {
+            return line.equals("----") || line.equals("");
+        }
+
+        private boolean processConditionalRecord(String line, SqlLogicTestRuntimeData runtimeData)
+        {
+            String[] splitLine = line.split(" ");
+            if (splitLine.length <= 1) {
+                return false;
+            }
+            DatabaseDialect dialect;
+            String potentialDialect = splitLine[1];
+            try {
+                dialect = DatabaseDialect.valueOf(potentialDialect.toUpperCase());
+                if (line.startsWith("onlyif")) {
+                    runtimeData.setCurrStatementsOnlyIfDialect(dialect);
+                    return true;
+                }
+                if (line.startsWith("skipif")) {
+                    runtimeData.addToSkipIfSet(dialect);
+                    return true;
+                }
+            }
+            catch (IllegalArgumentException e) {
+                LOG.debug("This line had a skip-if or only-if dialect that isn't supported: %s", line);
+            }
+            return false;
+        }
+
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+        {
+            return CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                throws IOException
+        {
+            if (isHidden(file)) {
+                return CONTINUE;
+            }
+            if (result.getNumFilesVisited() > maxNumFilesToVisit) {
+                return TERMINATE;
+            }
+            try (BufferedReader reader = newBufferedReader(file, defaultCharset())) {
+                SqlLogicTestRuntimeData runtimeData = new SqlLogicTestRuntimeData();
+                while (true) {
+                    final Optional<String> line = Optional.ofNullable(reader.readLine());
+                    if (!line.isPresent()) {
+                        break;
+                    }
+                    if (processConditionalRecord(line.get(), runtimeData)) {
+                        continue;
+                    }
+                    if (isLineBeforeStatement(line.get())) {
+                        if (statementIsAcceptableToCurrDialect(runtimeData)) {
+                            processStatement(reader);
+                        }
+                        runtimeData.reset();
+                    }
+                }
+            }
+            result.incrementNumFilesVisited();
+            return CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFileFailed(Path file, IOException exc)
+        {
+            LOG.warn("Failed to access file: " + file.toString());
+            return CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult postVisitDirectory(Path dir, IOException exc)
+        {
+            return CONTINUE;
+        }
+    }
+
+    private static class SqlLogicTestRuntimeData
```

Comment:
Maybe you can define an object SqlLogicTestRecord and this information can live in there, since it is only associated with a particular record. It would be cleaner then calling reset() after every record. Another advantage of that approach is you can parse in one place instead of passing BufferedReader into static helpers.

Reply:
Done.